microcosm_fastapi/namespaces.py

```diff
@@ -39,3 +39,17 @@ def path_prefix(self):
     @property
     def subject_name(self):
         return name_for(self.subject)
+
+    def extract_hostname_from_request(self, request: Request):
+        url = str(request.url)
+        return url.split('/api/')[0]
+
+    def url_for(self, request: Request, operation: Operation, **kwargs):
```

Comment:
Trying to replicate the `microcosm_flask` `url_for` method here - very hacky though..